integrations/gitlab_v1/main.py

```diff
@@ -0,0 +1,52 @@
+from typing import AsyncGenerator, Dict, Any
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+from fastapi import Request
+from loguru import logger
+from gitlab_integration import GitLabIntegration, ObjectKind
+
+
+gitlab_integration = GitLabIntegration()
+
+
+@ocean.router.post("/webhook/gitlab")
+async def gitlab_webhook(request: Request):
+    payload = await request.json()
+    event_type = payload.get("object_kind")
+
+
+    await gitlab_integration.handle_webhook_event(event_type, payload)
+    return {"status": "success"}
+
+
+@ocean.on_resync(ObjectKind.GROUP)
+async def on_resync_groups(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    async for item in gitlab_integration.resync_resources(ObjectKind.GROUP):
+        yield item
```

Comment:
can you also have a generic resync so that it can query the gitlab API?

@ocean.on_resync()
async def on_resync_globals(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
    async for item in gitlab_integration.resync_resources(kind):
        yield item